### ISSUE TITLE
Accessibility issues on Paginator/DataTable and ConfirmationDialog

### ISSUE THREAD

**Issue by mandy6720** [primefaces/primeng#2731]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**
Please fork the plunkr below and create a case demonstrating your bug report. Issues without a plunkr have much less possibility to be reviewed.

http://plnkr.co/edit/D1AK7i

**Current behavior**

Not really a bug, per se, but Paginator, DataTable and Modal do not meet standards for accessibility (WCAG/508 compliance).

Specifically:

-- lack of meaningful paginator alt text
When using a screen reader, the buttons for the paginator do not have meaningful text to indicate what their function is to a visually impaired user. 
(Suggestion: add aria-labels)

-- sortable columns do not have focus 
A user using keyboard/tab navigation is unable to tab to (focus) the sort arrows on a sortable column - they are unable to toggle the sort order in the DataTable. This doesn't necessarily make the table unusable for a visually impaired user, but would be a nice to have.
(Suggestion: add a tabIndex=0 to the arrow icon with an aria-label describing the sort direction)

-- focus on selectable row (shorthand selection column in DataTable)
When using the shorthand/build-in select column, a user using keyboard/tab navigation is unable to tab to/navigate to the checkboxes in that column.
(Suggestion: add a tabIndex=0 to the checkbow inputs)

-- focus on pop-up buttons (modal)
When a modal is opened, focus should go to the first interactable element on the modal window - otherwise, the user would have to tab through all of the elements on the current page before getting to the modal window.
(Suggestion: when the window is summoned, use .focus() or similar to programmatically set the focus to the upper-right close button or other interactable element)

**Expected behavior**

1. All buttons on paginator would have meaningful text for screenreaders.

2. Sort arrows for columns would be tab-able and have meaningful text.

3. The build-in selection column's checkboxes would be included in the tab navigation.

4. When a modal window is opened, focus would move to the upper righthand close button (x) or first other interactable element.

**Minimal reproduction of the problem with instructions**

Go to http://plnkr.co/edit/aBJgQZ?p=preview and click somewhere within the preview (right hand) frame. Using screen reading software (such as JAWS), navigate through the page using the keyboard (tab navigation).

**What is the motivation / use case for changing the behavior?**

Using primeNG in a government project which requires any web-apps/sites to be 508 compliant

**Please tell us about your environment:**

Operating system: OS Sierra
IDE: Visual Studio Code
package manager: npm
HTTP server: Node (from Angular CLI v1.0.0-rc.4)

* **Angular version:** 2.0.X

2.4.10

* **PrimeNG version:** 2.0.X

2.0.4

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

All
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5] 
all

* **Node (for AoT issues):** `node --version` =   v6.10.2

**Comment by chris-jones-pixitmedia** [comment]
+1

**Comment by cagataycivici** [comment]
Please create separate issue tickets for each case, we're constantly improving accessibility and can do this but need each case in its own ticket to manage it easily.

**Comment by jason-jason-h** [comment]
Was this ported to 6?
